### ISSUE TITLE
How to swipe on LongClick?

### ISSUE THREAD

**Issue by vorsichtdiekurve** [h6ah4i/android-advancedrecyclerview#118]
Is there a simple possibility to enable swiping only after longclick, like it is possible with the drag and drop functionality? Do i need to override some listeners? Thanks in advance for any help.

**Comment by h6ah4i** [comment]
Currently, there is no way to implement such behavior except for modifying the library. I'll add a new method in next release.

**Comment by h6ah4i** [comment]
Hi. I have just published v0.8.0. It introduced a new flag which indicates swiping starts on long press.

```java
    @Override
    public int onGetSwipeReactionType(MyViewHolder holder, int position, int x, int y) {
        return Swipeable.REACTION_CAN_SWIPE_LEFT | Swipeable.REACTION_MASK_START_SWIPE_LEFT |
                Swipeable.REACTION_CAN_SWIPE_RIGHT | Swipeable.REACTION_MASK_START_SWIPE_RIGHT |
                Swipeable.REACTION_START_SWIPE_ON_LONG_PRESS;
    }
```

- `REACTION_CAN_SWIPE_(LEFT | RIGHT)`
 - -> can swipe left or right
- `REACTION_MASK_START_SWIPE_(LEFT | RIGHT)`
 - -> but it does not start even if finger moves to left or right
- `REACTION_START_SWIPE_ON_LONG_PRESS`
 - -> start swiping on long press

ref.) https://github.com/h6ah4i/android-advancedrecyclerview/blob/f007b52c6e196e29c6142f7aa92791d6f520eb19/example/src/main/java/com/h6ah4i/android/example/advrecyclerview/demo_s_longpress/MySwipeableItemAdapter.java#L166

**Comment by OmarAliSaid** [comment]
Can you please provide a documentation on how to use the library .. Im new
in android and actually i cant understand the exambles you provide under
usage at github
On Oct 3, 2015 10:01 AM, "Haruki Hasegawa" <notifications@github.com> wrote:
>
> Hi. I have just published v0.8.0. It introduced a new flag which
indicates swiping starts on long press.
>
>     @Override
>     public int onGetSwipeReactionType(MyViewHolder holder, int position,
int x, int y) {
>         return Swipeable.REACTION_CAN_SWIPE_LEFT |
Swipeable.REACTION_MASK_START_SWIPE_LEFT |
>                 Swipeable.REACTION_CAN_SWIPE_RIGHT |
Swipeable.REACTION_MASK_START_SWIPE_RIGHT |
>                 Swipeable.REACTION_START_SWIPE_ON_LONG_PRESS;
>     }
>
> REACTION_CAN_SWIPE_(LEFT | RIGHT)
> -> can swipe left or right
> REACTION_MASK_START_SWIPE_(LEFT | RIGHT)
> -> but it does not start even if finger moves to left or right
> REACTION_START_SWIPE_ON_LONG_PRESS
> -> start swiping on long press
>
> ref.)
https://github.com/h6ah4i/android-advancedrecyclerview/blob/f007b52c6e196e29c6142f7aa92791d6f520eb19/example/src/main/java/com/h6ah4i/android/example/advrecyclerview/demo_s_longpress/MySwipeableItemAdapter.java#L166
>
> —
> Reply to this email directly or view it on GitHub.

--

**Comment by h6ah4i** [comment]
@OmarAliSaid Hi. 

Have you used the `RecyclerView` (just in a basic use)? If not, write a small exercise app first. I think [Vogella's tutorial](http://www.vogella.com/tutorials/AndroidRecyclerView/article.html) is a good reference.

Then, I have just wrote a tutorial about how to add swipeable feature to your adapter. Try it next.

- https://github.com/h6ah4i/android-advancedrecyclerview/wiki/Tutorial---How-to-add-SWIPEABLE-feature-to-your-adapter

Thanks.
